### DISCUSSION TITLE
How can I set a chart to always show description by default?

### DISCUSSION THREAD

**Discussion by rjlopes** [apache/superset#18347] [Q&A / Help] [answered]
If I define a description for a chart I can manually use the "Toggle chart description" to show the description for a given chart.

How can I define on a per chart or global basis if the description should be visible de default?

![image](https://user-images.githubusercontent.com/457848/141980300-e5b33cb3-5d2a-42a4-898f-5641e964edbb.png)

![image](https://user-images.githubusercontent.com/457848/141980319-ee45a026-7b3b-494f-811b-63b04ff8ed8d.png)

![image](https://user-images.githubusercontent.com/457848/141980356-93235b9f-72d2-40d1-8dce-c1b2d29b7f43.png)

**Comment by DanielFonteneleNogueira** [comment]
Excellent question. I'm in the same problem.

**Comment by cooley-pe** [comment]
I would like this feature as well. Without it, stakeholders have to know:
* That this toggle options exists
* That the chart has a description
These descriptions would be a great way to add context, but I can't trust that users will see them, making them significantly less valuable.

**Comment by cooley-pe** [comment]
I got the advice that the `expand_slices` key in the dashboard properties controls this. But there doesn't seem to be a way to set it (i.e. `Save As` doesn't work) https://github.com/apache/superset/issues/8791

At the very least it would be nice to know the format for manually adding them:
```
{
  "timed_refresh_immune_slices": [],
  "expanded_slices": {<format?>},
  "refresh_frequency": 0,
  "default_filters": "{}",
...
}
```

**Reply by cooley-pe** [reply, marked as answer]
> I got the advice that the `expand_slices` key in the dashboard properties controls this. But there doesn't seem to be a way to set it (i.e. `Save As` doesn't work) #8791
> 
> At the very least it would be nice to know the format for manually adding them:
> 
> ```
> {
>   "timed_refresh_immune_slices": [],
>   "expanded_slices": {<format?>},
>   "refresh_frequency": 0,
>   "default_filters": "{}",
> ...
> }
> ```

Ok so it looks like the format is:
```
"expand_slices": {"<slice_id>": true}
```
I can confirm that I was able to get this working.

**Comment by sfirke** [comment]
This works, but in case it affects others, sometimes the description is hidden after I save the dashboard.  After I tell it to show chart description once, it stays expanded permanently.

There's a related feature request here that would make this a little easier: https://github.com/apache/superset/discussions/22724

**Reply by sfirke** [reply]
Update: in 6.0.0rc3 this behavior of sometimes hiding chart descriptions despite this setting seems to be fixed. The feature request is still a valid, new thing.
